electrum/tests/regtest/regtest.sh

```diff
@@ -356,3 +356,19 @@ if [[ $1 == "watchtower" ]]; then
     echo "watchtower publishes justice transaction"
     wait_until_spent $ctx_id 1  # alice's to_local gets punished immediately
 fi
+
+if [[ $1 == "unixdomain" ]]; then
+    # This looks different because it has to run the entire daemon
+    # Test domain socket behavior
+    ./run_electrum --regtest daemon -d --unixdomain # Start daemon with unix domain socket
+    ./run_electrum --regtest stop # Errors if it can't connect
+    # Test custom socket path
+    f=$(mktemp --dry-run)
+    ./run_electrum --regtest daemon -d --unixdomain --unixsockpath=$f
+    [ -S $f ] # filename exists and is socket
+    ./run_electrum --regtest stop
+    rm $f # clean up
+    # Test for regressions in the ordinary TCP functionality.
+    ./run_electrum --regtest daemon -d --unixdomain # Start daemon with Internet socket
```

Comment:
> Isn't this using a Unix socket?

Oops!

>  Also, can we explicitly test for whether the Unix socket file is gone here like it should be?

Why should it be gone? I think that might create some problems of its own. Just spitballing here, but if an attacker can cause me to use a certain socket, that would also enable them to make me to delete that socket. That seems like not desirable behavior at all, potentially.

Under "standard" conditions, it will always use the same socket file, so it doesn't really have to be cleaned up. And if a caller designates a new socket file to use, it's the caller's task to clean it up. 

At the end of the day, I don't want to add complexity in detecting whether we created that socket file or it existed before and some extra config about that, and I don't want the undesired behavior of going around and deleting random socket files, so the current behaviour seems fine.

Reply:
Oh, I forgot about this thread, sorry.
@JeremyRand, could you clarify if the socket file should be cleaned up by the daemon upon stopping - and if so, what is the purpose of doing that? (note that if the daemon got killed for whatever reason it couldn't delete the socket file in any case)